File: app/src/main/java/com/esoxjem/movieguide/AppModule.java
package com.esoxjem.movieguide;

import android.app.Application;
import android.content.Context;
import android.content.res.Resources;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import io.realm.Realm;

/**
 * @author arunsasidharan
 * @author pulkitkumar
 */
@Module
public class AppModule {
    private Context context;

    AppModule(Application application) {
        context = application;
    }

    @Provides
    @Singleton
    public Context provideContext() {
        return context;
    }

    @Provides
    @Singleton
    public Resources provideResources(Context context) {
        return context.getResources();
    }

    @Provides
    @Singleton
    public Realm provideRealm() {
        return Realm.getDefaultInstance();
    }
}

File: app/src/main/java/com/esoxjem/movieguide/AppModule.kt
package com.esoxjem.movieguide

import android.app.Application
import android.content.Context
import android.content.res.Resources

import javax.inject.Singleton

import dagger.Module
import dagger.Provides

/**
 * @author arunsasidharan
 * @author pulkitkumar
 */
@Module
class AppModule constructor(application: Application) {
    private val context = application

    @Provides
    @Singleton
    fun provideContext(): Context {
        return context
    }

    @Provides
    @Singleton
    fun provideResources(context: Context): Resources {
        return context.resources
    }
}
